cumulusci/tasks/salesforce/users/permsets.py

```diff
@@ -29,65 +31,98 @@ def _init_options(self, kwargs):
         if user[self.assignment_child_relationship]:
             assigned_perms = {
                 r[self.assignment_lookup]
                 for r in user[self.assignment_child_relationship]["records"]
             }
+        return assigned_perms
 
-        # Find Ids for requested Perms
+    def _get_perm_ids(self):
         api_names = "', '".join(self.options["api_names"])
         perms = self.sf.query(
             f"SELECT Id,{self.permission_name_field} FROM {self.permission_name} WHERE {self.permission_name_field} IN ('{api_names}')"
         )
-        perms = {p[self.permission_name_field]: p["Id"] for p in perms["records"]}
-
-        # Raise for missing perms
-        for api_name in self.options["api_names"]:
-            if api_name not in perms:
-                raise CumulusCIException(
-                    f"{self.permission_label} {api_name} was not found."
+        perms_by_ids = {
+            p["Id"]: p[self.permission_name_field] for p in perms["records"]
+        }
+
+        missing_perms = [
+            api_name
+            for api_name in self.options["api_names"]
+            if api_name not in perms_by_ids.values()
+        ]
+        if missing_perms:
+            raise CumulusCIException(
+                f"The following {self.permission_label}s were not found: {', '.join(missing_perms)}."
+            )
+        return perms_by_ids
+
+    def _get_assignments(self, user_id, assigned_perms):
+        assignments = []
+        for perm, perm_name in self.perms_by_id.items():
+            if perm not in assigned_perms:
+                self.logger.info(
+                    f'Assigning {self.permission_label} "{perm_name}" to {user_id}.'
                 )
-
-        # Assign all not already assigned
-        for api_name in self.options["api_names"]:
-            if perms[api_name] not in assigned_perms:
-                self.logger.info(f'Assigning {self.permission_label} "{api_name}".')
                 assignment = {
-                    "AssigneeId": user["Id"],
+                    "attributes": {"type": self.assignment_name},
+                    "AssigneeId": user_id,
+                    self.assignment_lookup: perm,
                 }
-                assignment[self.assignment_lookup] = perms[api_name]
-                # Create the new assignment.
-                getattr(self.sf, self.assignment_name).create(assignment)
+                assignments.append(assignment)
             else:
                 self.logger.warning(
-                    f'{self.permission_label} "{api_name}" is already assigned.'
+                    f'{self.permission_label} "{perm_name}" is already assigned to {user_id}.'
                 )
+        return assignments
+
+    def _insert_assignments(self, records_to_insert):
+        request_body = json.dumps({"allOrNone": False, "records": records_to_insert})
+        result = self.sf.restful("composite/sobjects", method="POST", data=request_body)
```

Comment:
Hitting the 200-row limit here seems unlikely but definitely possible. Could we do a check first to avoid an inscrutable exception?